src/redisearch_rs/low_memory_thin_vec/src/lib.rs

```diff
@@ -1313,6 +1352,52 @@ impl<T: Clone> LowMemoryThinVec<T> {
     }
 }
 
+impl<T: Copy + std::fmt::Debug> LowMemoryThinVec<T> {
+    /// Reserves capacity to fit the given `prefix` slice,
+    /// moves the current elements back to make room for the prefix
+    /// and copies the prefix to the front of the vector.
+    pub fn prepend_with_slice(&mut self, prefix: &[T]) {
```

Comment:
`prepend_slice`, perhaps, rather than `prepend_with_slice`? 

Feel free to ignore the suggestion.

Reply:
I felt `prepend_slice` is a bit ambiguous, so I'm keeping it as is.